Index: kodiak-sources-4.6.6/arch/arm/mach-msm/smd_init_dt.c
===================================================================
--- kodiak-sources-4.6.6.orig/arch/arm/mach-msm/smd_init_dt.c
+++ kodiak-sources-4.6.6/arch/arm/mach-msm/smd_init_dt.c
@@ -21,7 +21,7 @@
 #include <linux/io.h>
 
 #include <mach/msm_ipc_logging.h>
-#include <smd_private.h>
+#include "smd_private.h"
 
 #define MODULE_NAME "msm_smd"
 #define IPC_LOG(level, x...) do { \
Index: kodiak-sources-4.6.6/drivers/cpufreq/cpufreq_interactive.c
===================================================================
--- kodiak-sources-4.6.6.orig/drivers/cpufreq/cpufreq_interactive.c
+++ kodiak-sources-4.6.6/drivers/cpufreq/cpufreq_interactive.c
@@ -801,7 +801,7 @@ static ssize_t show_target_loads(
 		ret += sprintf(buf + ret, "%u%s", target_loads[i],
 			       i & 0x1 ? ":" : " ");
 
-	ret += sprintf(buf + --ret, "\n");
+	sprintf(buf + ret - 1, "\n");
 	spin_unlock_irqrestore(&target_loads_lock, flags);
 	return ret;
 }
@@ -844,7 +844,7 @@ static ssize_t show_above_hispeed_delay(
 		ret += sprintf(buf + ret, "%u%s", above_hispeed_delay[i],
 			       i & 0x1 ? ":" : " ");
 
-	ret += sprintf(buf + --ret, "\n");
+	sprintf(buf + ret - 1, "\n");
 	spin_unlock_irqrestore(&above_hispeed_delay_lock, flags);
 	return ret;
 }
Index: kodiak-sources-4.6.6/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.c
===================================================================
--- kodiak-sources-4.6.6.orig/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.c
+++ kodiak-sources-4.6.6/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.c
@@ -658,7 +658,7 @@ uint32_t ov680_set_flash_current_on_eof(
 	int flash_update = 0;
 	uint64_t prev_frame_num = 0;
 
-	uint32_t flash_current[4];
+	uint32_t flash_current[4] = {};
 
 	// Go through the list and look for a frame that requires a current update
 	for(i=0;i<OV680_MAX_CURRENT_BUFFER;i++)
Index: kodiak-sources-4.6.6/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.h
===================================================================
--- kodiak-sources-4.6.6.orig/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.h
+++ kodiak-sources-4.6.6/drivers/media/platform/msm/camera_v2/sensor/ov680/ov680.h
@@ -8,7 +8,7 @@
 #define OV680_H
 
 #include <mach/board.h>
-#include <ov680_config.h>
+#include "ov680_config.h"
 #include <linux/firmware.h>
 #include "ov680_firmware.h"
 #include "ov680_flash.h"
Index: kodiak-sources-4.6.6/kernel/timeconst.pl
===================================================================
--- kodiak-sources-4.6.6.orig/kernel/timeconst.pl
+++ kodiak-sources-4.6.6/kernel/timeconst.pl
@@ -370,9 +370,9 @@ if ($hz eq '--can') {
 	}
 
 	@val = @{$canned_values{$hz}};
-	if (!defined(@val)) {
+	#if (!defined(@val)) {
 		@val = compute_values($hz);
-	}
+	#}
 	output($hz, @val);
 }
 exit 0;
Index: kodiak-sources-4.6.6/drivers/video/msm/mdss/mdss_mdp_trace.h
===================================================================
--- kodiak-sources-4.6.6.orig/drivers/video/msm/mdss/mdss_mdp_trace.h
+++ kodiak-sources-4.6.6/drivers/video/msm/mdss/mdss_mdp_trace.h
@@ -17,7 +17,7 @@
 #undef TRACE_SYSTEM
 #define TRACE_SYSTEM mdss
 #undef TRACE_INCLUDE_PATH
-#define TRACE_INCLUDE_PATH .
+#define TRACE_INCLUDE_PATH ../../drivers/video/msm/mdss/
 #undef TRACE_INCLUDE_FILE
 #define TRACE_INCLUDE_FILE mdss_mdp_trace
 
Index: kodiak-sources-4.6.6/drivers/video/msm/mdss/mdss_mdp.c
===================================================================
--- kodiak-sources-4.6.6.orig/drivers/video/msm/mdss/mdss_mdp.c
+++ kodiak-sources-4.6.6/drivers/video/msm/mdss/mdss_mdp.c
@@ -913,7 +913,7 @@ static ssize_t psr_read(struct device *d
 	ssize_t ret = 0;
 	int len = 0;
 
-	len = snprintf(buf, sizeof(buf), "%d\n", psr_user_enabled);
+	len = snprintf(buf, sizeof(*buf), "%d\n", psr_user_enabled);
 	ret = len;
 	return ret;
 }
Index: kodiak-sources-4.6.6/net/bluetooth/hci_conn.c
===================================================================
--- kodiak-sources-4.6.6.orig/net/bluetooth/hci_conn.c
+++ kodiak-sources-4.6.6/net/bluetooth/hci_conn.c
@@ -404,7 +404,7 @@ void hci_le_ltk_reply(struct hci_conn *c
 	memset(&cp, 0, sizeof(cp));
 
 	cp.handle = cpu_to_le16(conn->handle);
-	memcpy(cp.ltk, ltk, sizeof(ltk));
+	memcpy(cp.ltk, ltk, sizeof(*ltk));
 
 	hci_send_cmd(hdev, HCI_OP_LE_LTK_REPLY, sizeof(cp), &cp);
 }
Index: kodiak-sources-4.6.6/sound/soc/msm/qdsp6v2/rtac.c
===================================================================
--- kodiak-sources-4.6.6.orig/sound/soc/msm/qdsp6v2/rtac.c
+++ kodiak-sources-4.6.6/sound/soc/msm/qdsp6v2/rtac.c
@@ -25,7 +25,7 @@
 #include <sound/q6asm-v2.h>
 #include <sound/q6afe-v2.h>
 #include <sound/apr_audio-v2.h>
-#include <q6voice.h>
+#include "q6voice.h"
 #include "audio_acdb.h"
 
 
Index: kodiak-sources-4.6.6/Makefile
===================================================================
--- kodiak-sources-4.6.6.orig/Makefile
+++ kodiak-sources-4.6.6/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 4
 SUBLEVEL = 0
-EXTRAVERSION =
+EXTRAVERSION = -perf-g280c96c
 NAME = Saber-toothed Squirrel
 
 # *DOCUMENTATION*
