diff --git a/Makefile b/Makefile
index 52889d7a..eecc8f59 100755
--- a/Makefile
+++ b/Makefile
@@ -9,13 +9,9 @@ EXE = .exe
 RM = del
 endif
 
-CFLAGS = -ffunction-sections -O3
-
 UNAME_S := $(shell uname -s)
 ifeq ($(UNAME_S),Darwin)
     LDFLAGS += -Wl,-dead_strip
-else
-    LDFLAGS += -Wl,--gc-sections
 endif
 
 all:libmincrypt.a mkbootimg$(EXE) unpackbootimg$(EXE)
@@ -26,19 +22,19 @@ libmincrypt.a:
 	make -C libmincrypt
 
 mkbootimg$(EXE):mkbootimg.o
-	$(CROSS_COMPILE)$(CC) -o $@ $^ -L. -lmincrypt $(LDFLAGS) -s
+	$(CROSS_COMPILE)$(CC) -o $@ $^ -L. -lmincrypt $(LDFLAGS)
 
 mkbootimg-static$(EXE):mkbootimg.o
-	$(CROSS_COMPILE)$(CC) -o $@ $^ -L. -lmincrypt $(LDFLAGS) -static -s
+	$(CROSS_COMPILE)$(CC) -o $@ $^ -L. -lmincrypt $(LDFLAGS) -static
 
 mkbootimg.o:mkbootimg.c
 	$(CROSS_COMPILE)$(CC) -o $@ $(CFLAGS) -c $< -I. -Werror
 
 unpackbootimg$(EXE):unpackbootimg.o
-	$(CROSS_COMPILE)$(CC) -o $@ $^ $(LDFLAGS) -s
+	$(CROSS_COMPILE)$(CC) -o $@ $^ $(LDFLAGS)
 
 unpackbootimg-static$(EXE):unpackbootimg.o
-	$(CROSS_COMPILE)$(CC) -o $@ $^ $(LDFLAGS) -static -s
+	$(CROSS_COMPILE)$(CC) -o $@ $^ $(LDFLAGS) -static
 
 unpackbootimg.o:unpackbootimg.c
 	$(CROSS_COMPILE)$(CC) -o $@ $(CFLAGS) -c $< -Werror
diff --git a/libmincrypt/Makefile b/libmincrypt/Makefile
index a5fd66b3..8170888c 100755
--- a/libmincrypt/Makefile
+++ b/libmincrypt/Makefile
@@ -8,7 +8,6 @@ RM = del
 CP = copy /y
 endif
 
-CFLAGS = -ffunction-sections -O3
 EXT = a
 LIB = libmincrypt.$(EXT)
 LIB_OBJS = dsa_sig.o p256.o p256_ec.o p256_ecdsa.o rsa.o sha.o sha256.o
